src/main/scala/firrtl/analyses/IRLookup.scala

```diff
@@ -0,0 +1,343 @@
+// See LICENSE for license details.
+
+package firrtl.analyses
+
+import firrtl.annotations.TargetToken._
+import firrtl.annotations._
+import firrtl.ir._
+import firrtl.passes.MemPortUtils
+import firrtl.{DuplexFlow, ExpKind, SinkFlow, Flow, InstanceKind, Kind, SourceFlow, MemKind, PortKind, RegKind, UnknownFlow, Utils, WInvalid, WireKind}
+
+import scala.collection.mutable
+
+object IRLookup {
+  def apply(circuit: Circuit): IRLookup = ConnectionGraph(circuit).irLookup
```

Comment:
When I wrote this, I couldn't think of a better way to construct the IRLookup independently from the ConnectionGraph, yet still guaranteed to be in sync with a connection graph if you did want it. I think in a future PR we could potentially improve this. No action needed, besides maybe a note?

Reply:
Sure, in my next PRs, I'll find some other way to construct `IRLookup`.

Reply:
In my next PRs I think I should rewrite IRLookup construction.